pkg/koordlet/statesinformer/states_noderesourcetopology.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statesinformer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/metriccache"
+)
+
+func (s *statesInformer) syncNodeResourceTopology(node *corev1.Node) {
+	topologyName := fmt.Sprintf("koord-%s", node.Name)
+	ctx := context.TODO()
+	blocker := true
+	_, err := s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Get(ctx, topologyName, metav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			klog.Errorf("failed to get node resource topology %s, err: %v", topologyName, err)
+			return
+		}
+		topology := &v1alpha1.NodeResourceTopology{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: topologyName,
+				Labels: map[string]string{
+					"app.kubernetes.io/managed-by": "Koordinator",
+				},
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion:         "v1",
+						Kind:               "Node",
+						Name:               node.Name,
+						UID:                node.GetUID(),
+						Controller:         &blocker,
+						BlockOwnerDeletion: &blocker,
+					},
+				},
+			},
+		}
+		_, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Create(ctx, topology, metav1.CreateOptions{})
```

Comment:
the fields `topologyPolicies` and `zones` are required. This creation operation will fail. It is recommended to add an E2E test or verify it by yourself.